File: graph/model/keyword.go
package model

type Keyword struct {
	Value  string `json:"value"`
	Usages int64  `json:"usages"`
}

type SetKeywords struct {
	UUIDs     []string `json:"uuids"`
	Keywords  []string `json:"keywords"`
	Exclusive bool     `json:"exclusive"`
}

type KeywordLink interface {
	IsKeywordLink()
}

File: src/model/keyword.rs
#[derive(GraphQLObject)]
#[graphql(description="Keywords which link related content together")]
pub struct Keyword {
    pub value: String,
    pub usages: i32
}

#[derive(GraphQLInputObject)]
#[graphql(description="Ensure links from uuids to keywords. Exclusive removes all other links")]
pub struct SetKeywords {
    pub uuids: Vec<String>,
    pub keywords: Vec<String>,
    pub exclusive: bool
}
